asn1/InterledgerProtocol.asn

```diff
@@ -27,4 +28,21 @@ InterledgerProtocolPaymentMessage ::= SEQUENCE {
     }
 }
 
+InterledgerProtocolError ::= SEQUENCE {
+    -- Standardized error code
+    code IA5String (SIZE (3)),
+    -- Corresponding error name
+    codeName IA5String,
```

Comment:
> If this is being transmitted in binary what's the point of including the name? You need a parser anyway so that parser might as well give you the human-readable name

There may be new names your implementation isn't aware of. 

> Also should this field just be called name instead?

:+1:

> What should implementations do if the codeName doesn't match the expected one?

Show the name from the error. I'm not sure implementations *reading* the error would even consult a lookup table for the error names. The name is just for the developer's benefit and it makes sense to have it be generated by the party that generates the error for the reason above.